dGame/dMission/Mission.cpp

```diff
@@ -422,11 +422,18 @@ void Mission::YieldRewards() {
     }
 
     if (info->LegoScore > 0) {
-        characterComponent->SetUScore(characterComponent->GetUScore() + info->LegoScore);
-
-        if (info->isMission) {
-            GameMessages::SendModifyLEGOScore(entity, entity->GetSystemAddress(), info->LegoScore, 2);
+        eLootSourceType lootSource = info->isMission ? LOOT_SOURCE_MISSION : LOOT_SOURCE_ACHIEVEMENT;
+        int rewardedLegoScore = info->LegoScore;
+        // Define 124370 somewhere.
+        if(characterComponent->GetUScore() >= 124370) {
```

Comment:
Should probably define this in [CharacterComponent.h](https://github.com/DarkflameUniverse/DarkflameServer/blob/main/dGame/dComponents/CharacterComponent.h) via `static const uint64_t MaxUScore = 1234370`

Reply:
Actually, I think it should look that up in CDClient/CDServer via https://docs.lu-dev.net/en/latest/database/LevelProgressionLookup.html i.e. https://explorer.lu/misc/level-progression

Reply:
That makes sense. Then CharacterComponent could have a `MaxLevel` const defined as `45`?

Reply:
> That makes sense. Then CharacterComponent could have a `MaxLevel` const defined as `45`? Or is that defined elsewhere?

Yes, something like that. I think this might be defined here: https://explorer.lu/api/v0/tables/WorldConfig/1

Reply:
And the `LevelCapCurrencyConversion` set to 21 is there as well. I just have no clue whether WorldConfigID = 1 is set somewhere

Reply:
I think assuming 1 is fine, if it's the only config there is? Or shove it into `Game.h` 😬

so tl;dr

- take coins-per-level (`LevelCapCurrencyConversion`) and max level (`LevelCap`) from CDServer `WorldConfig` table, where `WorldConfigID = 1`
- take xp amount from CDServer `LevelProgressionLookup` table where `id = ${LevelCap}`

Reply:
Noted.  I havent worked directly with the CDServer table yet so i'll need to familiarize myself with it.  Should be able to get this done before eod though.  On a side note, what would a rebase commit look like?  I understand what it does but don't know how to tell if I have or have not done that.

Reply:
Good question! A rebase commit doesn't exist in the same way a merge does. Rebase is when you tell git to re-create the commits you made (same name, same author) for a new base (new changes on main), which, if it succeeds, results in the same list of commits you had before just with different hashes. you then need to force-push that to overwrite the old list because they're technically not the same.

Reply:
Is what I did below correct?  I see the same two reverts that I did on accident last night but with different hashes

Reply:
I think that was a "normal" merge of the branch, but into itself... as long as you keep a separate copy of your chaged file, I think you can try to figure it out with this PR without losing any progress. The basic idea is to run

    git checkout main

to go to the main branch

    git pull

to make sure it's up to date

    git checkout NameOfThisBranch

to switch (back) to this branch

    git rebase main

to do the rebase

    git push -f

to update this pull request

Reply:
I believe I did it correctly!  See below and feel free to let me know if it was incorrect.